File: src/lib/src/main/java/work/lclpnet/ap2/api/util/scoreboard/CustomScoreboardObjective.java
package work.lclpnet.ap2.api.util.scoreboard;

import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.numbers.NumberFormat;
import net.minecraft.server.level.ServerPlayer;
import org.jetbrains.annotations.Nullable;

public interface CustomScoreboardObjective {

    void setScore(String scoreHolder, int score);

    void setDisplayName(String scoreHolder, @Nullable Component display);

    void setNumberFormat(String scoreHolder, NumberFormat numberFormat);

    void removeEntry(String scoreHolder);

    default void setScore(ServerPlayer player, int score) {
        setScore(player.getScoreboardName(), score);
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/util/scoreboard/CustomScoreboardObjective.kt
package work.lclpnet.ap2.util.scoreboard

import net.minecraft.network.chat.Component
import net.minecraft.network.chat.numbers.NumberFormat
import net.minecraft.server.level.ServerPlayer

interface CustomScoreboardObjective {

    fun setScore(scoreHolder: String, score: Int)

    fun setDisplayName(scoreHolder: String, display: Component?)

    fun setNumberFormat(scoreHolder: String, numberFormat: NumberFormat?)

    fun removeEntry(scoreHolder: String)

    fun setScore(player: ServerPlayer, score: Int) {
        setScore(player.scoreboardName, score)
    }
}
